

 boehm-gc/pthread_support.c (GC_get_nprocs): Use sysconf as fallback.


Index: boehm-gc/pthread_support.c
===================================================================
--- boehm-gc/pthread_support.c	(revision 127207)
+++ boehm-gc/pthread_support.c	(working copy)
@@ -724,7 +724,8 @@
     f = open("/proc/stat", O_RDONLY);
     if (f < 0 || (len = STAT_READ(f, stat_buf, STAT_BUF_SIZE)) < 100) {
 	WARN("Couldn't read /proc/stat\n", 0);
-	return -1;
+	/* Fallback to sysconf after the warning */
+	return sysconf(_SC_NPROCESSORS_ONLN);
     }
     for (i = 0; i < len - 100; ++i) {
         if (stat_buf[i] == '\n' && stat_buf[i+1] == 'c'
